### PR TITLE
[Merged by Bors] - feat: port `Order.Zorn`

### PR DESCRIPTION
**Pull request by ADedecker**

### CODE REVIEW

File: Mathlib/Order/Zorn.lean [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+Fleuriot, Tobias Nipkow, Christian Sternagel.
+-/
+
+
+open Classical Set
+
+variable {α β : Type _} {r : α → α → Prop} {c : Set α}
+
+-- Porting note: the following line was produced by mathport (?)
+-- mathport name: «expr ≺ »
```
Comment (dupuisf) [inline Mathlib/Order/Zorn.lean:74]:
```suggestion
```
I think we normally just delete that stuff.

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

**Comment by hrmacbeth** [comment]
It failed to build, apparently complaining about the local notation already being used in another file:
```text
./././Mathlib.lean:1:0: error: import Mathlib.Order.Zorn failed, environment already contains
'«term_≺__3»._closed_6._cstage2' from Mathlib.Algebra.Order.EuclideanAbsoluteValue
```
Strange because this file successfully built before.  I don't know if the problem was in deleting the comment
```lean
-- mathport name: «expr ≺ »
```
or in merging master.  I'll try merging master first.

**Comment by ADedecker** [comment]
I've encountered this before, this is a bug (see [here](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Local.20notation.20conflict)). The temporary fix is to give a name to the notation. I'm not at my computer right now, but you can have a look at what I did in `Order.Antichain` to see the syntax for naming notations.

**Pull request closed without merging**